### ISSUE TITLE
Wayland support (exit with: Unable to load OpenGL entry points)

### ISSUE THREAD

**Issue by alexpattyn** [flathub/org.freeorion.FreeOrion#3]
# Specs

Fedora 33 (Beta) Silverblue
Windowing system: Wayland
Gnome: 3.38.0 
Freeorion: 0.4.10
GPU: Radeon RX 580 Series (POLARIS10, DRM 3.38.0, 5.8.12-300.fc33.x86_64, LLVM 11.0.0)
CPU: AMD® Ryzen 5 1600 six-core processor × 12 

# Issue

When trying to launch the program it fails with the error below. 

## Output
```
$ flatpak run org.freeorion.FreeOrion 
Unable to load OpenGL entry points.

GLEW reported:
Unknown error
terminate called after throwing an instance of '(anonymous namespace)::QuitSignal'
```

**Comment by travier** [comment]
I have just updated to the latest release but did not try on Wayland.
  - Can you try with the previous build? (See https://github.com/flatpak/flatpak/wiki/Tips-&-Tricks#downgrading).
  - Can you also make sure to run a `flatpak update` again to install any potentially missing platform extensions/drivers?

**Comment by travier** [comment]
https://github.com/flatpak/flatpak/wiki/Tips-&-Tricks#downgrading

**Comment by travier** [comment]
Can you also make sure to run a `flatpak update` again to install any potentially missing platform extensions/drivers?

**Comment by Siosm** [comment]
Confirmed this is happening on Wayland only. Will see if this is specific to this Flatpak or if this does not work with upstream either.

**Comment by Iolaum** [comment]
Hit the same bug on Fedora Silverblue 33 on a laptop with i5-7300h cpu (and nvidia 1050Ti on Optimus configuration running on nouveau drivers).

**Comment by travier** [comment]
I'm sorry but I have not found the time to work on that yet. Any help is appreciated.

**Comment by Iolaum** [comment]
I don't have the skills to help with this unfortunately. All I can say is that selecting Xorg, (instead of Wayland) on the Gnome logging screen is a way to sidestep the problem (the game loads fine in the resulting Xorg session).

P.S. Thank you for all the work you 've done packaging the game @travier

**Comment by travier** [comment]
Can you give the build from https://github.com/flathub/org.freeorion.FreeOrion/pull/4 a try to see if this is a valid workaround until the issue is fixed? Thanks!

**Comment by Iolaum** [comment]
Yes it is. I was able to start the game within a Wayland session. :+1:

**Comment by travier** [comment]
I've merged the temporary fix while we figure out Wayland support upstream.

**Comment by travier** [comment]
This is somewhat solved with https://github.com/flathub/org.freeorion.FreeOrion/pull/16. The application still runs under Xwayland however as far as I can see.

**Comment by travier** [comment]
Found the following topic discussing this issue: https://freeorion.org/forum/viewtopic.php?t=12419
